src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -262,21 +263,24 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
                 {
                     InitializeHub(hub, connection);
 
-                    var result = methodInfo.Invoke(hub, invocationDescriptor.Arguments);
-                    var resultTask = result as Task;
-                    if (resultTask != null)
+                    object result = null;
+                    if (methodExecutor.IsMethodAsync)
                     {
-                        await resultTask;
-                        if (methodInfo.ReturnType.GetTypeInfo().IsGenericType)
+                        if (methodExecutor.TaskGenericType == null)
                         {
-                            var property = resultTask.GetType().GetProperty("Result");
-                            invocationResult.Result = property?.GetValue(resultTask);
+                            await (Task)methodExecutor.Execute(hub, invocationDescriptor.Arguments);
```

Comment:
ExecuteAsync calls TaskOfTActionExecutorAsync which assumes you have a Task<T> return type. I would be okay with refactoring it.